### ISSUE TITLE
[bug] 'cmake' generator does not set INTERFACE_LINK_DIRECTORIES

### ISSUE THREAD

**Issue by slietzau** [conan-io/conan#10837]
### Environment Details (include every applicable attribute)
  * Conan version: 1.46.0

### Steps to reproduce (Include if Applicable)

The error surfaced when trying to consume  `boost` with option `magic_autolink=True`. This option causes the link library list to be empty. Instead the boost headers include `#pragma comment` statements to select the libraries to link against. For this approach to work the libraries need to be found by the compiler.

Here are the locations for both cmake generators ([cmake](https://github.com/conan-io/conan/blob/e98a7590b544e738fdb927678d34821ccb78ad78/conans/client/generators/cmake_common.py#L166) and [cmake_find_package](https://github.com/conan-io/conan/blob/8f7ac4306cdf7fc4536728b07457b20895be0023/conans/client/generators/cmake_find_package.py#L179)). The latter correctly sets the `INTERFACE_LINK_DIRECTORIES` variable while the former is missing that part.

**Comment by memsharded** [comment]
Hi @slietzau 

The ``cmake`` and ``cmake_find_package`` generators are no longer being developed, and efforts have been shifted to the new ``CMakeDeps`` and ``CMakeToolchain`` integration. Please have a look at those and report if necessary.

**Comment by slietzau** [comment]
@memsharded Thanks for the quick response. `CMakeDeps` + `CMakeToolchain` don't work either.

**Comment by memsharded** [comment]
can you clarify why you need that property? Aren't the targets defined by CMakeDeps working? I recommend starting with an example with ``conan new hello/0.1 -m=cmake_lib`` to report (The ``test_package/conanfile.py`` is using ``CMakeDeps`` there)

**Comment by slietzau** [comment]
Thanks again. I just checked the boost recipe (because we use a modified version of the `boost` package). Turns out we clear the libs for all components:

```diff
- self.cpp_info.components[module].libs = module_libraries
+ self.cpp_info.components[module].libs = [] if self.options.magic_autolink else module_libraries
```

Anyway, the idea was to use an empty link list and let the compiler find the libraries. However that needs the `INTERFACE_LINK_DIRECTORIES` setting which is currently only supported by the `cmake_find_package` generator.

However, I guess it's not that good of an idea to remove all libraries from the link list. 

Here is a minimal working example based on `cmake_lib` template. I'm not yet experienced with conan `components` so that part might be more verbose than it needed to be. Worth noting is, that only the `cmake_find_package` generator for components includes the `INTERFACE_LINK_DIRECTORIES` property.

```diff
diff --git a/conanfile.py b/conanfile.py
index 83325ae..7811f22 100644
--- a/conanfile.py
+++ b/conanfile.py
@@ -42,4 +42,6 @@ class HelloConan(ConanFile):
         cmake.install()
 
     def package_info(self):
-        self.cpp_info.libs = ["hello"]
+        self.cpp_info.components['main'].includedirs.append('include')
+        self.cpp_info.components['main'].libdirs.append('lib')        
+        self.cpp_info.components['main'].libs = []
diff --git a/include/hello.h b/include/hello.h
index 9e87180..49f9d73 100644
--- a/include/hello.h
+++ b/include/hello.h
@@ -2,6 +2,7 @@

 #ifdef _WIN32
 #define hello_EXPORT __declspec(dllexport)
+#pragma comment(lib, "hello")
 #else
 #define hello_EXPORT
 #endif
diff --git a/test_package/CMakeLists.txt b/test_package/CMakeLists.txt
index c33c2a7..2df5beb 100644
--- a/test_package/CMakeLists.txt
+++ b/test_package/CMakeLists.txt
@@ -4,4 +4,4 @@ project(PackageTest CXX)
 find_package(hello CONFIG REQUIRED)

 add_executable(example src/example.cpp)
-target_link_libraries(example hello::hello)
+target_link_libraries(example hello::main)
```

With these changes the `CMakeDeps` generator cannot build the project. Switching to `cmake_find_package` builds the project.

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/pull/10984 implemented a new property to define this, with ``cpp_info.set_property("cmake_set_interface_link_directories", True)`` in the package recipe (the dependency, not the consumer).

This will be available next Conan 1.48.
